openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -1037,4 +1044,51 @@ private static SemanticModel getSemanticModel(Path clientPath) throws ProjectExc
             throw new ProjectException(e.getMessage());
         }
     }
+
+    public static String getOpenAPIType(Schema<?> schema) {
+        if (schema.getTypes() != null && !schema.getTypes().isEmpty()) {
+            for (String type : schema.getTypes()) {
+                // this returns the first non-null type in the list
+                if (!type.equals(NULL)) {
+                    return type;
+                }
+            }
+        } else if (schema.getType() != null) {
+            return schema.getType();
+        }
+        return null;
+    }
+
+    public static boolean isaArraySchema(Schema schema) {
+        return getOpenAPIType(schema) != null && Objects.equals(getOpenAPIType(schema), ARRAY);
+    }
+
+    public static boolean isaMapSchema(Schema schema) {
+        return schema.getAdditionalProperties() != null;
+    }
+
+    public static boolean isaObjectSchema(Schema schema) {
+        return getOpenAPIType(schema) != null && Objects.equals(getOpenAPIType(schema), OBJECT);
```

Comment:
Shall we add a condition for the scenario when properties are available without setting the type `object` since it is also mapped to object behaviour?

Reply:
I only replaced `schema instanceOf ObjectSchema` with `isObjectSchema` function. In the paces we have explicitly schecked for `schema.getProperties() != null` I have kept it as it is considering that there can be scenarios where the object is not defined as `type: object` but has properties.